### ISSUE TITLE
Bug: [ast-spec] wrong type for callee of CallExpression, NewExpression, TaggedTemplateExpression

### ISSUE THREAD

**Issue by Josh-Cena** [typescript-eslint/typescript-eslint#9229]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

ast-spec

### Playground Link

https://typescript-eslint.io/play/#ts=5.4.5&showAST=ts&fileType=.tsx&code=BQQwBAPhYEYJTDgbgFADsCmB3MpLXkVTyljgANzUg&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tiacTJTIAhtEK0yHJgBNK%2BSpPRQA7iKaRwYAL4hNQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
(a || b)();
new (a || b)();
(a || b)``;
```

### Expected Result

The callee should have type `Expression`

### Actual Result

The callee has type `LeftHandSideExpression` which doesn't include `BinaryExpression`. This is probably a remnant from when we had `ParenthesizedExpression`.

Also see estree: https://github.com/estree/estree/blob/master/es5.md#callexpression, https://github.com/estree/estree/blob/master/es2015.md#taggedtemplateexpression

### Versions

Everything latest
